src/CodeReview.Orchestrator/Services/DockerVolumeImporter.cs

```diff
@@ -39,7 +45,10 @@ public async Task ImportVolumesAsync(IProcessingContext context, params Volume[]
                     if (!_directoryService.Exists(folderPath))
                         continue;
 
-                    await using var outStream = _tarArchiveService.Create(folderPath);
+                    Stream outStream;
+                    using var _ = _directoryService.DirectorySize(folderPath) <= ByteSize.FromMegaBytes(300)
```

Comment:
I suggest to create static variable (i doubt we can create constant) which describes 300 MB limit. E.g. name it like `MaxInMemoryArchiveSize` or similar descriptive name. If there is a reference to corresponding article or documentation it would be good to put a link.